kong/db/schema/entities/cas.lua

```diff
@@ -0,0 +1,41 @@
+local typedefs = require "kong.db.schema.typedefs"
+local openssl_x509 = require "openssl.x509"
+local ngx_time = ngx.time
+
+return {
+  name        = "cas",
+  primary_key = { "id" },
+
+  fields = {
+    { id             = typedefs.uuid, },
+    { created_at     = typedefs.auto_timestamp_s },
+    { cert           = typedefs.certificate { required = true }, },
+    { tags           = typedefs.tags },
+  },
+
+  entity_checks = {
+    { custom_entity_check = {
+      field_sources = { "cert", },
+      fn = function(entity)
+        local cert = openssl_x509.new(entity.cert)
+        local not_before, not_after = cert:getLifetime()
+        local now = ngx_time()
+
+        if not_before > now then
+          return nil, "certificate unusable, \"Not Before\" time is " ..
+                      "in the future"
+        end
+
+        if not_after < now then
```

Comment:
for this one I can see the obvious application, though I always get a "what about clock skew" kneejerk reaction when I see things like this. :) But I can't think up of a problem scenario right now (even considering short-lived certificates), and it adds user-friendliness. :+1:

Reply:
Yeah. Regardless the reason a CA that has expired is no use for Kong, so this check definitely helps with avoiding some of the surprises.